### PR TITLE
optimizer/plan: fix join panic

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala @hanfei19910905  PTAL

File: optimizer/plan/planbuilder_join.go [hunk 1]
```diff
@@ -224,7 +224,10 @@ func (p *joinPath) containsTable(table *ast.TableName) bool {
 		return p.table == table
 	}
 	if p.subquery != nil {
-		return p.asName.L == table.Name.L
+		if table != nil {
```
Comment (shenli) [inline optimizer/plan/planbuilder_join.go]: Check if table == nil at the beginning of the function?

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**